jobs/grafana/templates/create-update-datasources.erb

```diff
@@ -45,23 +26,29 @@ if [[ $exitCode != 0 ]]
 then
   echo -e "\nError: Cannot connect to Grafana after 3 retries."
   exit 1
-fi
+  fi
+
+<%
+  require 'json'
+  defaults = {'access' => 'proxy', 'basicAuth' => false}
+  datasources = p('grafana.datasources')
+  if_p('grafana.datasource') do |datasource|
+    datasources << p('grafana.datasource')
+  end
+%>
 
 <% datasources.each do |datasource| %>
 echo -e "\nCreating/Updating datasource '<%= datasource['name'] %>' at $(date)"
 
 DATASOURCE_NAME='<%= datasource['name'] %>'
 
-DATA='
-{"name":"'"${DATASOURCE_NAME}"'",
- "type":"<%= datasource['database_type'] %>",
- "access":"proxy",
- "url":"<%= datasource['url'] %>",
- "password":"<%= datasource['password'] %>",
- "user":"<%= datasource['user'] %>",
- "database":"<%= datasource['database_name'] %>",
- "basicAuth":false
-}'
+<%
+  data = defaults.merge(datasource)
+  data['name'] = data.delete('database_name')
```

Comment:
Agreed, we actually *can* rename all these fields to match the datastore api--I just renamed them incorrectly earlier 😖. I just reverted to mapping over datasources to set default values and handle backwards compatibility, but this time with the fields named correctly. And I changed the description in `spec` to match the datastore api.